dt-mapping/location-grid-meta.php

```diff
@@ -128,6 +147,19 @@ public static function add_location_grid_meta( $post_id, array $location_grid_me
                 return new WP_Error( __METHOD__, 'Unable to create location_grid post meta and retrieve a key.' );
             }
 
+            if ( isset( $post_id ) ) {
+                $grid = self::get_existing_locations( $post_id, $location_grid_meta['grid_id'] );
+
+                dt_write_log( 'get_existing_locations -> grid' );
+                dt_write_log( $grid );
+
+                if( !empty( $grid ) )
+                {
+                    dt_write_log( 'Location already exists.' );
+                    return new WP_Error( __METHOD__, 'This location already exists in the contact.' );
```

Comment:
Should this really throw an error? Or should it just silently fail and log that the location wasn't added because it already exists? What context is this function being used in and what will happen in either the API request or UI when this function throws an error?

Reply:
Ok. That makes sense. I saw it returned an error and thought it threw it, but I guess each of those functions that call it can handle that error in their own way. Sounds good. Thanks for clarifying. We should probably try to test those different functions that use this as best as we can.